modules/core/src/stat.simd.hpp

```diff
@@ -48,12 +48,16 @@ int normHamming(const uchar* a, int n)
 #  if defined CV_POPCNT_U64
         for(; i <= n - 8; i += 8)
         {
-            result += (int)CV_POPCNT_U64(*(uint64*)(a + i));
+            uint64_t val;
+            std::memcpy(&val, a + i, sizeof(val));
```

Comment:
Do we need extra code path under alignment check to avoid possible perf regressions?

Reply:
On x86_64 and AArch64 the machine code will be the same (https://godbolt.org/z/4sMoE8cWG), on ARMv7 and RISC-V it could be less efficient. IMO we can keep the simple code for now and then try to optimize it if it would be necessary.

It might be better to add separate scalar function `<typename T> int popcnt64(T * ptr)` instead of this macro and maybe sum bits for each byte instead of switching between aligned/unaligned load for specific platforms.